### PR TITLE
Task6

### PR DESCRIPTION
**Pull request by Xelsoid**

### CODE REVIEW

File: src/controllers/cart.controller.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import { isUserExists } from "../servises/user.service"
+import {
+    returnCartTotal,
+    updateUserCart,
+    returnUserCartData,
+    deleteUserCart
+} from "../servises/cart.service";
+import { createNewOrder } from '../servises/order.service'
+
+const app = express();
```
Comment (ShaxxKh) [inline src/controllers/cart.controller.ts]: there must be a single app in `src/app.ts` file that configured for all entities (cart, order, users)

in this `controller` file you should just have the callbacks like : `async (req: Request, res: Response, next: NextFunction) => {}` for routes. So you export all your controllers, import them in `src/app.ts` and you them like: `app.get('CART_ENDPOINTS.PROFILE_CART', getCart)`

File: src/controllers/cart.controller.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(logger);
+
+const validateCartRequest = (req: Request, res: Response, next: NextFunction) => {
+    const userId = req.header('x-user-id');
+
+    if(!userId) {
```
Comment (ShaxxKh) [inline src/controllers/cart.controller.ts]: move this and the check below to separate `auth.ts` file

File: src/controllers/cart.controller.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    if(!updatedCart || !cartTotal){
+        throw(new Error("Internal Server error"));
+    }
+
+    res.status(200).send({
+        "data": {
+            "cart": updatedCart,
+            "total": cartTotal
+        },
+        "error": null
```
Comment (ShaxxKh) [inline src/controllers/cart.controller.ts]: this is not actually necessary

File: src/controllers/index.controller.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import express from 'express';
```
Comment (ShaxxKh) [inline src/controllers/index.controller.ts]: remove this file

**Review by ShaxxKh — CHANGES_REQUESTED** [review]

File: src/controllers/cart.controller.ts (hunk 1 above)
Comment (Xelsoid) [reply src/controllers/cart.controller.ts]: done, really helpful advice

File: src/controllers/cart.controller.ts (hunk 2 above)
Comment (Xelsoid) [reply src/controllers/cart.controller.ts]: done

File: src/controllers/cart.controller.ts (hunk 3 above)
Comment (Xelsoid) [reply src/controllers/cart.controller.ts]: removed

File: src/controllers/index.controller.ts (hunk 4 above)
Comment (Xelsoid) [reply src/controllers/index.controller.ts]: removed

**Comment by Xelsoid** [comment]
> not finalized yet

finalized

**Review by ShaxxKh — APPROVED** [review]
good job

**Pull request merged**